sources/osgShadow/ShadowMap.js

```diff
@@ -780,37 +781,61 @@ define( [
             var light = lightSource.getLight();
             var camera = this._cameraShadow;
 
+            var worldLightPos = this._worldLightPos;
+            var worldLightDir = this._worldLightDir;
+
             // make sure it's not modified outside our computations
             // camera matrix can be modified by cullvisitor afterwards...
+
             Matrix.copy( camera.getProjectionMatrix(), this._projectionMatrix );
             Matrix.copy( camera.getViewMatrix(), this._viewMatrix );
             var projection = this._projectionMatrix;
             var view = this._viewMatrix;
 
             // inject camera world matrix.
-            // from light current world/pos
+            // from light current world/pos and camera eye pos.
+            var lightMatrix = light.getPositionnedMatrix();
```

Comment:
Yes but you added methods in light that only make sense in your context. So
it s more appropriate to take this from cull visitor.
But in general what is computed in cull visitor Computed data about
transform from cull visitor should stay in cull / draw visitor

On Wed, Apr 15, 2015, 18:24 Tuan Kuranes <notifications@github.com> wrote:

> In sources/osgShadow/ShadowMap.js
> <https://github.com/cedricpinson/osgjs/pull/296#discussion_r28436326>:
>
> >              Matrix.copy( camera.getProjectionMatrix(), this._projectionMatrix );
> >              Matrix.copy( camera.getViewMatrix(), this._viewMatrix );
> >              var projection = this._projectionMatrix;
> >              var view = this._viewMatrix;
> >
> >              // inject camera world matrix.
> > -            // from light current world/pos
> > +            // from light current world/pos and camera eye pos.
> > +            var lightMatrix = light.getPositionnedMatrix();
>
> isn't light::getPositionnedMatrix exaclty that ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cedricpinson/osgjs/pull/296/files#r28436326>.
>